test/cppsuite/test_harness/workload_generator.h

```diff
@@ -66,46 +66,36 @@ class workload_generator : public component {
     void
     run()
     {
-        configuration *transaction_config, *update_config, *insert_config;
-        int64_t min_operation_per_transaction, max_operation_per_transaction, read_threads,
-          update_threads, value_size;
+        configuration *read_config, *update_config, *insert_config;
 
         /* Populate the database. */
         _database_operation->populate(_database, _timestamp_manager, _config, _tracking);
         _db_populated = true;
 
         /* Retrieve useful parameters from the test configuration. */
-        transaction_config = _config->get_subconfig(OPS_PER_TRANSACTION);
         update_config = _config->get_subconfig(UPDATE_CONFIG);
         insert_config = _config->get_subconfig(INSERT_CONFIG);
-        read_threads = _config->get_int(READ_THREADS);
-        update_threads = _config->get_int(UPDATE_THREADS);
-
-        min_operation_per_transaction = transaction_config->get_int(MIN);
-        max_operation_per_transaction = transaction_config->get_int(MAX);
-        testutil_assert(max_operation_per_transaction >= min_operation_per_transaction);
-        value_size = _config->get_int(VALUE_SIZE);
-        testutil_assert(value_size >= 0);
+        read_config = _config->get_subconfig(READ_CONFIG);
 
         /* Generate threads to execute read operations on the collections. */
-        for (size_t i = 0; i < read_threads && _running; ++i) {
-            thread_context *tc = new thread_context(_timestamp_manager, _tracking, _database,
-              thread_operation::READ, max_operation_per_transaction, min_operation_per_transaction,
-              value_size, throttle());
+        for (size_t i = 0; i < read_config->get_int(THREAD_COUNT) && _running; ++i) {
```

Comment:
I personally dont like the call of a function in a loop even though it's probably optimized.

Reply:
I think its cleaner, and would be an inexpensive call so I'll leave it.